wasm/src/debug_service.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use qsc::fir::StmtId;
+use qsc::interpret::stateful;
+use qsc::interpret::{stateful::Interpreter, Value};
+use qsc::{PackageType, SourceMap};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use wasm_bindgen::prelude::*;
+
+use crate::{language_service::VSDiagnostic, CallbackReceiver};
+
+#[wasm_bindgen]
+pub struct DebugService {
+    interpreter: Interpreter,
+}
+
+#[wasm_bindgen]
+impl DebugService {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self {
+            interpreter: Interpreter::new(false, SourceMap::default(), PackageType::Lib)
+                .expect("Couldn't create interpreter"),
+        }
+    }
+
+    pub fn load_source(&mut self, path: &str, source: &str) -> bool {
+        let source_map = SourceMap::new([(path.into(), source.into())], None);
+        match Interpreter::new(true, source_map, qsc::PackageType::Exe) {
+            Ok(interpreter) => {
+                self.interpreter = interpreter;
+                self.interpreter.set_entry().is_ok()
+            }
+            Err(_) => false,
+        }
+    }
+
+    pub fn get_stack_frames(&self) -> JsValue {
+        let frames = self.interpreter.get_stack_frames();
+
+        let list = StackFrameList {
+            frames: frames
+                .iter()
+                .map(|s| StackFrame {
+                    name: format!("{} {}", s.name, s.functor),
+                    path: s.path.clone(),
+                    lo: s.lo,
+                    hi: s.hi,
+                })
+                .collect(),
+        };
+        serde_wasm_bindgen::to_value(&list).expect("failed to serialize stack frame list")
+    }
+
+    pub fn eval_continue(
+        &mut self,
+        event_cb: &js_sys::Function,
+        ids: &[u32],
+    ) -> Result<JsValue, JsValue> {
+        if !event_cb.is_function() {
+            return Err(JsError::new("Events callback function must be provided").into());
+        }
+        let bps: Vec<_> = ids.iter().map(|f| StmtId::from(*f)).collect();
+
+        match self.run_internal(
+            |msg: &str| {
+                // See example at https://rustwasm.github.io/wasm-bindgen/reference/receiving-js-closures-in-rust.html
+                let _ = event_cb.call1(&JsValue::null(), &JsValue::from(msg));
+            },
+            &bps,
+        ) {
+            Ok(None) => Ok(JsValue::UNDEFINED),
+            Ok(Some(v)) => Ok(JsValue::from(std::convert::Into::<usize>::into(v))),
```

Comment:
What is this `usize` that could be returned from running? Can you add a comment as to what this value represents? Thanks.